web-components/src/components/form/CreateSellOrderForm.tsx

```diff
@@ -0,0 +1,144 @@
+import React, { useEffect, useState } from 'react';
+import { Formik, Form, Field, FormikErrors } from 'formik';
+
+import AmountField from '../inputs/AmountField';
+import SelectTextField, { Option } from '../inputs/SelectTextField';
+
+import Submit from './Submit';
+import { requiredMessage, validateAmount } from '../inputs/validation';
+import { RegenModalProps } from '../modal';
+import NumberTextField from '../inputs/NumberTextField';
+import CheckboxLabel from '../inputs/CheckboxLabel';
+import { Subtitle } from '../typography';
+import { Box, useTheme } from '@mui/material';
+
+export interface CreateSellOrderProps {
+  batchDenoms: Option[];
+  sellDenom: string;
+  availableAmountByBatch: { [batchDenom: string]: number };
+  onSubmit: (values: FormValues) => Promise<void>;
+}
+
+interface FormProps extends CreateSellOrderProps {
+  onClose: RegenModalProps['onClose'];
+}
+
+export interface FormValues {
+  batchDenom?: string;
+  price?: number;
+  amount?: number;
+  disableAutoRetire?: boolean;
+}
+
+const CreateSellOrderForm: React.FC<FormProps> = ({
+  sellDenom,
+  batchDenoms,
+  availableAmountByBatch,
+  onClose,
+  onSubmit,
+}) => {
+  const [options, setOptions] = useState<Option[]>([]);
+  const theme = useTheme();
+
+  const initialValues = {
+    batchDenom: batchDenoms[0].value,
+    price: undefined,
+    amount: undefined,
+    disableAutoRetire: false,
+  };
+
+  useEffect(() => {
+    setOptions(batchDenoms);
+  }, [batchDenoms]);
+
+  const validateHandler = (values: FormValues): FormikErrors<FormValues> => {
+    let errors: FormikErrors<FormValues> = {};
+
+    if (!values.batchDenom) {
+      errors.batchDenom = requiredMessage;
+    }
+    if (!values.price) {
+      errors.price = requiredMessage;
+    }
+    const errAmount = validateAmount(
+      availableAmountByBatch[values.batchDenom ?? ''],
+      values.amount,
+    );
+    if (errAmount) errors.amount = errAmount;
+
+    return errors;
+  };
+
+  return (
+    <Formik
+      initialValues={initialValues}
+      validate={validateHandler}
+      onSubmit={onSubmit}
+    >
+      {({ values, submitForm, isSubmitting, isValid, submitCount, status }) => (
+        <Form>
+          <Field
+            name="batchDenom"
+            label="Batch denom"
+            component={SelectTextField}
+            options={options}
+            sx={{ mb: theme.spacing(10.5) }}
+          />
+          <Box
+            sx={{ display: 'flex', alignItems: 'end', mb: theme.spacing(0.5) }}
+          >
+            <Field
+              component={NumberTextField}
+              name="price"
+              label="Price"
+              increment={0.1}
+              min={0.0}
+              arrows={false}
+              sx={{ maxWidth: '238px' }}
```

Comment:
I"m not sure if this sx prop will work correctly

This is something I think we might want to discuss in an architecture call, but formik's `Field` component passes whatever props you give it to the underlaying component, regardless of whether they exist on that component, and i'm not sure if the `sx` props will be passed to anything here (most fields rely on `className`). It's also not typed well so the compiler won't show any issues.

We could correct this by changing the fields to use the formik context (`useFormikContext`) instead of relying one passed `FieldProps` values, which would give us type safety, but would require a decent amount of work

Reply:
I confirm that this is working. Reason is that all these `Field` use components from `formik-mui` which does extend props from mui so using `sx` here is fine.